### ISSUE TITLE
Stop YOLO development in the helm chart.

### ISSUE THREAD

**Issue by gothrek22** [1Password/scim-examples#244]
Hi,

It seems that the helm chart provided by 1Password has met some growing pains as part of it's development.

There is no proper changelog, default value changes and other random issues related to a disconnect between the chart and the bridge itself. [Here's a related issue in that repo](https://github.com/1Password/op-scim-helm/issues/126)

Our bridge has been down for the whole day due to an issue after upgrading to the latest version of the chart. We've been able to fix it only after setting the following in the chart:
- scim.config.tlsDomain: null
- manually editing the deployment and adding: OP_TLS_KEY_FILE, OP_TLS_CERT_FILE as empty, else the binary would stubbornly try to run in TLS mode
- redis.enabled: false

Only after setting the above and updating the key paths that changed were we able to have a functioning bridge.

I've raised a couple of issues in that repo, but it seems that it's not really a priority, I've also tried contacting support with no luck.

Hope this raises awareness about this issue.

**Comment by grellyd** [comment]
Hi @gothrek22,

Thank you for expressing your thoughts and concerns. 

As the 1Password SCIM helm chart matures, we have indeed experienced some growing pains. Up to this point, the Helm chart has mostly existed as a method of distributing nearly identical kubernetes deployment to our 1-Click applications in the [Google Cloud](https://github.com/1Password/op-scim-gcp-marketplace) and [Digital Ocean](https://github.com/1Password/marketplace-kubernetes) marketplaces. Bluntly, we err'd in marking the helm chart as version 2.x.x, which at the time seemed a natural extension after the helm chart had to diverge from 1Password SCIM bridge versions. A more appropriate version would have been 0.x.x given the developmental and fast changing nature of this chart. 

No matter the reasons, having your deployment break on an upgrade sucks. I apologise for our choices that caused you have such a terrible experience. 

With that in mind, here is what we are going to do going forward:
* Move to correct semantic versioning of the helm chart releases. You should be able to trust the version numbers.
* Introduce proper changelogs. This may be as simple as linking to the [SCIM bridge changelogs](https://app-updates.agilebits.com/product_history/SCIM), but we will work to ensure the changelog accurately reflects the changes to the helm chart as well. 
* Expand the documentation to better describe the intended usage patterns. 

I hope that addresses your concerns. I look forward to your thoughts.

Graham

**Comment by gothrek22** [comment]
@grellyd lovely, Thank you Graham, this is what I wanted to hear.

**Comment by grellyd** [comment]
Following up, these changes have been made in the `op-scim-helm` repo. The documentation still needs some improvement, but we have introduced proper release changelogs (https://github.com/1Password/op-scim-helm/releases) and are paying much closer attention to versioning. In addition we've made a few internal process changes to ensure these changes persist. 

![Screenshot 2023-08-21 at 2 45 44 PM](https://github.com/1Password/scim-examples/assets/7812612/27b7ea85-9c3c-47f4-81ec-4963666d246e)
